### PR TITLE
Add Google Drive Download API

### PR DESCRIPTION
**Pull request by mustafa1728**
Fixes https://github.com/pykale/pykale/issues/182.

### Description
Adds drive download function from [video loading example](https://github.com/pykale/pykale/blob/main/examples/video_loading/main.py) to core API for better reusability. API allows easier download of google drive files from their ids.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Comment by XianyuanLiu** [comment]
This example is grabbing a zip file from google drive. Could you add another example with no need to uncompress like URL one?

**Review by XianyuanLiu — APPROVED** [review]
Well done. LGTM. Thanks.

File: tests/utils/test_download.py [hunk 1]
```diff
@@ -3,14 +3,19 @@
...
 
 output_directory = Path().absolute().parent.joinpath("test_data/download")
 PARAM = [
     " https://github.com/pykale/data/raw/main/video_data/video_test_data/ADL/annotations/labels_train_test/adl_P_11_train.pkl;a.pkl;pkl",
     " https://github.com/pykale/data/raw/main/video_data/video_test_data.zip;video_test_data.zip;zip",
 ]
 
+GDRIVE_PARAM = [
+    "1U4D23R8u8MJX9KVKb92bZZX-tbpKWtga;demo_datasets.zip;zip",
+    "1SV7fmAnWj-6AU9X5BGOrvGMoh2Gu9Nih;dummy_data.csv;csv",
```
Comment (haipinglu) [inline tests/utils/test_download.py:16]: Where are these located? Which Google account? Permanent?

**Review by haipinglu — CHANGES_REQUESTED** [review]
Thanks. The video loading example should be updated to use this API and remove duplication. See the other question.

File: tests/utils/test_download.py (hunk 1 above)
Comment (mustafa1728) [reply tests/utils/test_download.py:16]: The first one was taken directly from the video loading example, and is in Raivo's Google account. The second one is in my account.

These are not permanent. Is there a PyKale google account? Maybe these dummy files can be added there.

**Comment by mustafa1728** [comment]
> The video loading example should be updated to use this API and remove duplication. 

So may I edit that example in this PR itself or would Raivo be doing that?

**Comment by haipinglu** [comment]
> > The video loading example should be updated to use this API and remove duplication.
> 
> So may I edit that example in this PR itself or would Raivo be doing that?

**Comment by haipinglu** [comment]
> > The video loading example should be updated to use this API and remove duplication.
> 
> So may I edit that example in this PR itself or would Raivo be doing that?

Please do it @mustafa1728. Thanks.

File: tests/utils/test_download.py (hunk 1 above)
Comment (haipinglu) [reply tests/utils/test_download.py:16]: Thanks for the clarification. It is fine then. We do not have a PyKale Google account (yet). No need to create just for dummy files.

**Review by haipinglu — APPROVED** [review]

**Pull request merged**